### PR TITLE
Rebrand modals

### LINKED ISSUE #457: Rebrand DAO Dashboard - Modals
Update styles of modals on the staking page based on the [new designs](https://www.figma.com/design/NG6QClnVrvPRZ5gJ3s0zen/%E2%9F%81--Api3-DAO---Branding-Update?node-id=8676-107801&node-type=section&m=dev). This includes
- Stake / Unstake / Deposit / Withdraw modals
- Unsupported chain modal

### PR DESCRIPTION
**Pull request by Anboias**
Closes #457

#### Button component:
- remove unnecessary Button div wrapper
- fix footer focus border on key navigation (see video below)
- add new variants (tertiary & text-blue)

#### Input component:
- update Input component to new Branding

#### Modal component:
- update main Modal component to new Branding

#### Modals:
- update Deposit / Withdraw / Stake / Unstake modals
- update Confirm Unstake modal
- update Unsupported chains modal

## Why change the Button?
The div around the Button was unnecessary and it was affecting the [New Proposal modal](https://www.figma.com/design/NG6QClnVrvPRZ5gJ3s0zen/%E2%9F%81--Api3-DAO---Branding-Update?node-id=8676-108149&m=dev)'s Help tooltip display.
The footer Button focus was plain wrong and I took this opportunity to fix it.

#### Before 

https://github.com/user-attachments/assets/8fc86700-d072-443e-a58b-2591bf6e7a70

#### Now

https://github.com/user-attachments/assets/0b817c0b-5df1-4663-9053-28a528afe002

### Deposit modal (Withdraw / Stake / Unstake are similar)

| Mobile | Desktop |
|--------|--------|
| image | image | 

### Confirm Unstake modal

| Mobile | Desktop |
|--------|--------|
| image | image| 

### Unsupported Network modal

| Mobile | Desktop |
|--------|--------|
| image | image |

### CODE REVIEW

File: src/components/modal/modal.tsx [hunk 1]
```diff
@@ -71,22 +80,51 @@ export const Modal = (props: ModalProps) => {
...
+      children: string;
+      tooltipText?: null;
+      size?: null;
+      noMargin?: boolean;
+    }
+  | {
+      children: string;
+      tooltipText: string;
+      size: 'large';
+      noMargin?: boolean;
```
Comment (peterjurco) [inline src/components/modal/modal.tsx:94]: It doesn't seem we are using this variant of `ModalHeader` anywhere. Can't we just drop it?

**Review by peterjurco — APPROVED** [review]
Solid work here 👏 Thanks also for the detailed PR description

File: src/components/modal/modal.tsx (hunk 1 above)
Comment (Anboias) [reply src/components/modal/modal.tsx:94]: No, it's actually there in preparation for the upcoming `New Proposal` modal 😊

Comment (peterjurco) [reply src/components/modal/modal.tsx:94]: Oh, right!

**Pull request merged**